lib/client/hot.js

```diff
@@ -63,13 +67,16 @@ function result(modules, appliedModules) {
 }
 
 function check() {
-  module.hot.check().then((modules) => {
+  module.hot.check(!!options.reload).then((modules) => {
```

Comment:
should this be `false` permenantly, since we're handling applying the hot update ourselves on line 72/94?

Reply:
I mix code from https://github.com/webpack/webpack/blob/v3.0.0/hot/dev-server.js inside `hot.js`, which is currently more inspired by https://github.com/webpack/webpack/blob/v3.0.0/hot/only-dev-server.js. 

This way we use the autoApply option of `module.hot.check` to skip error handling set in `hotOptions`, and instead catch apply errors the same way as `/only-dev-server.js.`

Other solutions could be to change hotOptions:
  - Add reload conditions inside `onUnaccepted` / `onDeclined` / `onErrored` methods
  - Or set `ignoreUnaccepted` / `ignoreDeclined` / `ignoreErrored` to false if `options.reload` is true, and catch module.hot.apply error

Don't know which one is better though

Reply:
My apologies, I'm having a hard time understanding what you're trying to say here. I don't think that we should be mixing the two approaches at all. I'm OK with the PR overall, but I think this line needs to be reverted until more testing is done. Unfortunately that's not automated yet and will have to be done on my end locally to verify. I'd be happy to merge this PR without this line change until I can run more tests.

Reply:
Without this line unfortunately it will not fix the issue. 
However, I've made a new commit with another method which doesn't involve setting `autoApply` to true in `module.hot.check()`:

I don't pass the `hotOptions` to `module.hot.apply` if `options.reload` is set to true. This way, `module.hot.apply` will throw if there is an issue so I can catch the module.hot.apply promise error and call a `window.location.reload` if it's necessary.

Reply:
I'm not sure that's a good fix either. It doesn't feel good to be changing the way options are passed to force an unhandled error to be thrown. Feels kind of hacky. I think we need to take a step back and I need to look at the original issue again for a proper fix.

Reply:
Well, the error is handled:
  - Add a log message to say the hot patch cannot be applied, and don't use messages from hotOptions that say we ignore the error, because we don't with options.reload set to true
  - Reload the page

It's the purpose of options.reload to force reloading the page when a hot patch has thrown an error and can't be applied